staging/src/k8s.io/apiserver/pkg/endpoints/filters/traces.go

```diff
@@ -48,6 +50,8 @@ func WithTracing(handler http.Handler, tp trace.TracerProvider) http.Handler {
 		if r.URL != nil {
 			trace.SpanFromContext(r.Context()).SetAttributes(semconv.HTTPTarget(r.URL.RequestURI()))
 		}
+		// Add auditID attribute if available. This helps us connect traces to audit log entries
+		trace.SpanFromContext(r.Context()).SetAttributes(attribute.String("audit-id", audit.GetAuditIDTruncated(r.Context())))
```

Comment:
nit: can we  just do `trace.SpanFromContext(r.Context())` once, and then set attributes on it as needed?

Reply:
Good idea, updated